apis/kahu/v1beta1/restore_types.go

```diff
@@ -0,0 +1,114 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RestoreSpec defines the desired state of Restore
+type RestoreSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// BackupName is backup CR name specified during backup
+	BackupName string `json:"backupName,omitempty"`
+
+	// IncludeNamespaces are set of namespaces name considered for restore
+	IncludeNamespaces []string `json:"includeNamespace,omitempty"`
+
+	// ExcludeNamespaces are set of namespace name should not get considered for restore
+	ExcludeNamespaces []string `json:"excludeNamespaces,omitempty"`
+
+	// IncludeResources are set of kubernetes resource name considered for restore
+	IncludeResources []string `json:"includeResources,omitempty"`
+
+	// ExcludeResources are set of kubernetes resource name should not get considered for restore
+	ExcludeResources []string `json:"excludeResources,omitempty"`
+
+	// LabelSelector are label get evaluated against resource selection
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+
+	// NamespaceMapping is mapping between backed up namespace name against restore namespace name
+	NamespaceMapping map[string]string `json:"namespaceMapping,omitempty"`
+
+	// IncludeClusterResource is a flag for considering cluster wider resource during restore
+	IncludeClusterResource bool `json:"includeClusterResource,omitempty"`
+
+	// ResourcePrefix gets prepended in each restored resource name
+	ResourcePrefix string `json:"resourcePrefix,omitempty"`
+}
+
+type RestorePhase string
+
+const (
+	RestorePhaseInit             RestorePhase = "New"
+	RestorePhaseFailedValidation RestorePhase = "FailedValidation"
+	RestorePhaseInProgress       RestorePhase = "InProgress"
+	RestorePhaseCompleted        RestorePhase = "Completed"
+	RestorePhasePartiallyFailed  RestorePhase = "PartiallyFailed"
+	RestorePhaseFailed           RestorePhase = "Failed"
+	RestorePhaseDeleting         RestorePhase = "Deleting"
+)
+
+// RestoreProgress expresses overall progress of restore
+type RestoreProgress struct {
+	// TotalItems is count of resource to be process
+	TotalItems int `json:"totalItems,omitempty"`
+
+	// ItemsRestored is count of resource got processed
+	ItemsRestored int `json:"itemsRestored,omitempty"`
+}
+
+// RestoreStatus defines the observed state of Restore
+type RestoreStatus struct {
```

Comment:
what about validation errors, warnings, errors, etc. ?

Reply:
Added Failure reasons 
Created an issue for validaion error as it should be handle in Webhooks #36